crates/metadata/src/specs.rs

```diff
@@ -888,6 +966,10 @@ where
     pub fn display_name(&self) -> &DisplayName<F> {
         &self.display_name
     }
+
+    pub fn new_from_ty(ty: <F as Form>::Type, display_name: DisplayName<F>) -> Self {
```

Comment:
Again I would make this the `new` constructor, and change the existing `new<T>` to e.g. `of_type<T>` or maybe `with_type<T>` as other constructors are called on this.